mimic/rest/maas_api.py

```diff
@@ -518,9 +518,21 @@ def delete_entity(self, request, tenant_id, entity_id):
         Delete an entity, all checks that belong to entity, all alarms that belong to those checks
         """
         entities = self._entity_cache_for_tenant(tenant_id).entities_list
+        previous_size = len(entities)
         entities[:] = [entity for entity in entities
                        if entity.id != entity_id]
 
+        if len(entities) == previous_size:
```

Comment:
Intuitively, if an entity was deleted the size of the list must have changed.

The convention for deletes in this plugin is to write a list comprehension for the new contents of the list and assign it to a slice of itself, like you see above. This ensures we don't iterate over a list while modifying it, which complicates reasoning and causes bugs. A slight downside of this approach is that you have to check the size of the list before and after if you want to know if the expression actually deleted anything.